### PR TITLE
Pretty JSON export

### PR DESCRIPTION
**Pull request by uroybd**
This PR implements RapidJSON encoding and an optional pretty output for highlight export and share.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9165)

### CODE REVIEW

**Comment by uroybd** [comment]
@pazos

Please look into it when you're free. The PR is relatively simple.

**Comment by pazos** [comment]
Please remove the setting, is not needed.

**Comment by uroybd** [comment]
> Please remove the setting, is not needed.

Done.

**Review by pazos — APPROVED** [review]
Thanks!

File: plugins/exporter.koplugin/target/json.lua [hunk 1]
```diff
@@ -1,5 +1,7 @@
-local json = require("json")
 local Device = require("device")
+local rapidjson = require("rapidjson")
+local _ = require("gettext")
```
Comment (Frenzie) [inline plugins/exporter.koplugin/target/json.lua:3]: I assume this should be removed again?
```suggestion
```

**Review by Frenzie — CHANGES_REQUESTED** [review]

File: plugins/exporter.koplugin/target/json.lua (hunk 1 above)
Comment (pazos) [reply plugins/exporter.koplugin/target/json.lua]: thanks! I missed that :+1:

**Pull request merged**